Specify category when uploading `sarif` files

wzieba opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->

Relates to: AINFRA-951

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR is a follow up to #14287 . After this PR, Lint results from `wear` module started overwriting these from `WooCommerce`. This PR fixes the problem by providing `category` when uploading the `sarif` file. 

Implementation details: https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/183.

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

Compare [trunk results](https://github.com/woocommerce/woocommerce-android/security/code-scanning) and these from [the this PR](https://github.com/woocommerce/woocommerce-android/security/code-scanning?query=is%3Aopen+pr%3A14360).

- [ ] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

wzieba committed:
Send Lint sarif files with tool name so they're not overridden

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/263">22.9</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

wzieba committed:
Update ci toolkit version

wzieba committed:
Update ci toolkit version

wzieba committed:
Improve sarif files category names

ParaskP7 reviewed (approved):
Nicely done @wzieba and thanks for the fix/improvement, I reviewed this as part of [BKCIPL#183](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/183) and everything LGTM (ish[^1]), I am approving to unblock, tentative to updating `CI_TOOLKIT` to a released version prior to merging this. 🚀

[^1]: See [review](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/183#pullrequestreview-3047002295) and the testing side of things.

wzieba committed:
Don't specify category for the main report

So there's no mismatch of "closed" issues comparing to previous, non-category sarif

wzieba committed:
Revert "Don't specify category for the main report"

This reverts commit 13e2cc93bd25a2d02d2e97a48bd46c98de5b107e.

wzieba committed:
Update ci-toolkit to stable `5.4.0`
